src/main/java/poort8/ishare/core/Authorisation.java

```
@@ -1,13 +1,130 @@
+
+        String clientAssertion = CreateClientAssertion(partyId);
+
+        HashMap<String, String> parameters = new HashMap<>();
+        parameters.put("grant_type", "client_credentials");
+        parameters.put("scope", "iSHARE");
+        parameters.put("client_id", GetConfig("ClientId"));
+        parameters.put("client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer");
+        parameters.put("client_assertion", clientAssertion);
+
+        String form = parameters.entrySet()
+                .stream()
+                .map(e -> e.getKey() + "=" + URLEncoder.encode(e.getValue(), StandardCharsets.UTF_8))
+                .collect(Collectors.joining("&"));
+
+        HttpClient client = HttpClient.newHttpClient();
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(url))
+                .headers("Content-Type", "application/x-www-form-urlencoded")
+                .POST(HttpRequest.BodyPublishers.ofString(form))
+                .build();
+        HttpResponse<?> response;
+
+        try {
+            response = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+            if (response.statusCode() != 200) {
+                System.out.printf("Could not get access token from %s%n", partyId);
+                throw new RuntimeException();
+            }
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+
+        TokenResponse tokenResponse = null;
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            tokenResponse = mapper.readValue(response.body().toString(), TokenResponse.class);
+        } catch (JsonProcessingException e) {
+            System.out.println("Could not get access token from API response: " + e);
+        }
+
+        if (tokenResponse == null) {
+            System.out.println("Could not get access token from API response");
+            return "";
```

Better to throw an exception?